pyiron_base/project/generic.py

```diff
@@ -661,20 +661,71 @@ def job_table(
         """
         auto_refresh_job_status (bool): will automatically reload job status by calling refresh_job_status() upon calling job_table
         """
-        if auto_refresh_job_status:
-            self.refresh_job_status()
-        return self.db.job_table(
-            sql_query=self.sql_query,
-            user=self.user,
-            project_path=self.project_path,
-            recursive=recursive,
-            columns=columns,
-            all_columns=all_columns,
-            sort_by=sort_by,
-            full_table=full_table,
-            element_lst=element_lst,
-            **kwargs,
-        )
+        if not isinstance(self.db, FileTable):
+            if auto_refresh_job_status:
+                self.refresh_job_status()
+            return self.db.job_table(
+                sql_query=self.sql_query,
+                user=self.user,
+                project_path=self.project_path,
+                recursive=recursive,
+                columns=columns,
+                all_columns=all_columns,
+                sort_by=sort_by,
+                full_table=full_table,
+                element_lst=element_lst,
+                **kwargs,
+            )
+        elif not auto_refresh_job_status:
+            return self.db.job_table(
+                sql_query=self.sql_query,
+                user=self.user,
+                project_path=self.project_path,
+                recursive=recursive,
+                columns=columns,
+                all_columns=all_columns,
+                sort_by=sort_by,
+                full_table=full_table,
+                element_lst=element_lst,
+                **kwargs,
+            )
+        else:
```

Comment:
The code in the FileTable currently only contains the database interface but does not depend on the queuing system. I am not sure if we want to mix both. I see the database classes as one level and then we can have a derived class which handles the communication with the queuing system.

Reply:
Then an alternative would be to add support for `FileTable` in `Project.refresh_job_status` which is implemented purely in terms of `get_item_by_id` and `set_job_status`.  (I'm assuming that this doesn't work right now and that that's the reason for this PR, but I had to guess this).

Reply:
The `refresh_job_status` functions iterates over one job at a time, while here I get all jobs and then update them all in the database. I agree that the database is not very clean and we definitely have to rework this in particular also in connection with the support of remote clusters. But unfortunately I currently do not have the time to do a major refactoring of that part.

Reply:
> The `refresh_job_status` functions iterates over one job at a time, while here I get all jobs and then update them all in the database. 

Move it into the `FileTable` then since it is specific to that class.  It doesn't matter much to me whether this mixes queuing and database interactions since they are tangled in practice anyway.

> But unfortunately I currently do not have the time to do a major refactoring of that part.

And no one will have time to rework it in the future.  This is why I was insistent in the strategy meeting that when adding new things we make at least an attempt to have maintainable and well founded abstractions.  It might save a day of work now, but it will cost the next developer a week to understand and refactor it.
If we can at least keep everything contained in either `FileTable` or `DatabaseAccess` then unifying them in the future will be much easier.

Reply:
I do not see that moving it to FileTable would make the code more readable, because then we would also have to move all the `refresh_job_status` part to `DatabaseAccess`.

Reply:
The other issue with moving it to the `FileTable` is that the `FileTable` is connected to the `state` object just like the `QueueAdapter` so we would have to do a major refactoring of the whole interface.